File: groa-mapping/processing_scripts/05_reproduce_figure_1.py
import os
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns

# Define paths
current_dir = os.path.dirname(os.path.abspath(__file__))
repo_root = os.path.abspath(os.path.join(current_dir, ".."))
DATA_DIR = os.path.join(repo_root, "data")
OUTPUT_DIR = os.path.join(repo_root, "outputs")

def reproduce_figure_1_improved():
    print("Loading data...")
    try:
        measurements = pd.read_csv(os.path.join(DATA_DIR, "biomass_litter_CWD.csv"), encoding='latin1')
        sites = pd.read_csv(os.path.join(DATA_DIR, "sites.csv"), encoding='latin1')
    except Exception as e:
        print(f"Error loading data: {e}")
        return

    # Merge Data
    data = pd.merge(measurements, sites, on="site.id")

    # Filter Data
    # Paper: "natural forest regrowth", Age <= 30
    # Variable: 'aboveground_biomass'
    plot_data = data[
        (data['stand.age'] > 0) & 
        (data['stand.age'] <= 30) & 
        (data['variables.name'] == 'aboveground_biomass')
    ].copy()

    # Convert to Carbon (approximate)
    plot_data['carbon'] = plot_data['mean_ha'] * 0.47

    # Assign Biomes (Improved Logic based on Climate)
    # Note: This is an approximation. The original paper used a spatial join with Dinerstein et al. 2017.
    
    def assign_biome(row):
        lat = abs(row['lat_dec'])
        amt = row['AMT'] # Temp
        amp = row['AMP'] # Precip
        
        # Handle missing climate data
        if pd.isna(amt) or pd.isna(amp):
            # Fallback to simple latitude
            if lat > 50: return "Boreal"
            if lat > 23.5: return "Temperate"
            return "Tropical Moist" # Default for tropics if unknown

        # Boreal
        if lat > 50 or amt < 3:
            return "Boreal"
        
        # Temperate
        if lat > 23.5:
            # Attempt to split Conifer vs Broadleaf? 
            # Very hard without species/region. 
            # Let's keep them grouped or try a rough precip split?
            # Paper: Temperate Conifer vs Temperate Broadleaf
            # Broadleaf often wetter/warmer in some contexts, but Conifer rainforests exist.
            # Let's group them as "Temperate (Broadleaf & Conifer)" to be honest about the proxy.
            return "Temperate" 
            
        # Tropical / Subtropical
        if lat <= 23.5:
            if amp < 1000:
                return "Tropical Dry"
            elif amp < 1500:
                return "Tropical Savanna"
            else:
                return "Tropical Moist"
        
        return "Other"

    plot_data['Biome_Proxy'] = plot_data.apply(assign_biome, axis=1)

    print(f"Plotting {len(plot_data)} points...")
    print("Biome counts:")
    print(plot_data['Biome_Proxy'].value_counts())

    # Plotting
    sns.set_theme(style="whitegrid")
    
    # Create a FacetGrid
    g = sns.lmplot(
        data=plot_data, 
        x="stand.age", 
        y="carbon", 
        col="Biome_Proxy", 
        col_wrap=3, # 3 columns to fit 5-6 plots nicely
        height=4, 
        aspect=1.0,
        scatter_kws={'alpha': 0.4, 's': 15, 'color': 'darkgreen'},
        line_kws={'color': 'black', 'linewidth': 1.5}
    )

    # Titles and Labels
    g.fig.suptitle("Figure 1 Reproduction: Carbon Accumulation by Biome (Climate Proxy)", y=1.02, fontsize=16)
    g.set_axis_labels("Stand Age (years)", "Plant Carbon (Mg C ha-1)")
    g.set_titles("{col_name}")
    
    # Save
    output_path = os.path.join(OUTPUT_DIR, "figure_1_reproduction_improved.png")
    plt.savefig(output_path, bbox_inches='tight', dpi=300)
    print(f"Figure saved to: {output_path}")

if __name__ == "__main__":
    reproduce_figure_1_improved()

File: groa-mapping/processing_scripts/05_reproduce_figure_1.R
# Script to reproduce Figure 1: Carbon Accumulation by Biome
# Note: The original paper used a spatial join with a Biome shapefile (Dinerstein et al. 2017) 
# to assign biomes to each site. Since that shapefile is not included in this repo,
# this script demonstrates the plotting logic using the available data.

# Check for required packages
if (!require("ggplot2")) install.packages("ggplot2")
if (!require("dplyr")) install.packages("dplyr")

library(ggplot2)
library(dplyr)

# 1. Load Data
# Adjust paths if running from a different directory
measurements <- read.csv("../data/biomass_litter_CWD.csv", stringsAsFactors = FALSE)
sites <- read.csv("../data/sites.csv", stringsAsFactors = FALSE)

# 2. Merge Data
# We need coordinates from 'sites' to (hypothetically) get biomes
data <- merge(measurements, sites, by = "site.id")

# 3. Filter Data
# Paper: "natural forest regrowth", Age <= 30
# 'refor.type' usually indicates restoration type. We'll assume 'NR' or similar is Natural Regrowth.
# Let's inspect unique refor.type if possible, but for now we filter by age.
# Variable: 'aboveground_biomass' (converted to Carbon) or 'total_plant_carbon' if available.
# The paper says they summed above+below. Here we just use what we have.

plot_data <- data %>%
  filter(stand.age > 0 & stand.age <= 30) %>%
  filter(variables.name == "aboveground_biomass") %>%
  mutate(carbon = mean_ha * 0.47) # Conversion to Carbon (approximate)

# 4. Assign Biomes (Mock Logic)
# Since we lack the Biome map, we will create a rough proxy based on Latitude
# to demonstrate the faceting used in Figure 1.
plot_data$Biome_Proxy <- case_when(
  abs(plot_data$lat_dec) > 50 ~ "Boreal",
  abs(plot_data$lat_dec) > 35 ~ "Temperate",
  abs(plot_data$lat_dec) > 23.5 ~ "Subtropical",
  TRUE ~ "Tropical"
)

# 5. Plot (Reproducing Figure 1 Style)
p <- ggplot(plot_data, aes(x = stand.age, y = carbon)) +
  geom_point(alpha = 0.5, color = "darkgreen") +
  geom_smooth(method = "lm", formula = y ~ x, color = "black") + # Linear fit as per paper
  facet_wrap(~ Biome_Proxy, scales = "free") +
  theme_bw() +
  labs(
    title = "Figure 1 Reproduction: Carbon Accumulation by Biome (Proxy)",
    subtitle = "Note: Biomes approximated by latitude as original shapefile is missing",
    x = "Stand Age (years)",
    y = "Plant Carbon (Mg C ha-1)"
  )

# Save Plot
ggsave("../outputs/figure_1_reproduction.png", plot = p, width = 10, height = 8)

print("Figure 1 reproduction saved to outputs/figure_1_reproduction.png")
